tests/test_framework/phpbb_functional_test_case.php

```diff
@@ -1433,4 +1433,112 @@ protected function get_hidden_fields($crawler, $url)
 
 		return $file_form_data;
 	}
+
+	/**
+	 * Get HTML of the crawler
+	 * See https://symfony.com/doc/current/components/dom_crawler.html#component-dom-crawler-dumping
+	 *
+	 * @param Symfony\Component\DomCrawler\Crawler $crawler Crawler instance
+	 * @param string $url Request URL
+	 *
+	 * @return array Hidden form fields array
+	 */
+	protected function dump_crawler($crawler)
+	{
+		if (!$crawler)
+		{
+			return;
+		}
+
+		$html = '';
+		foreach ($crawler as $domElement)
+		{
+			$html .= $domElement->ownerDocument->saveHTML($domElement);
+		}
+
+		return $html;
+	}
+
+	/**
+	 * Get username of currently logged in user
+	 *
+	 * @return string|bool username if logged in, false otherwise
+	 */
+	protected function get_logged_in_user()
+	{
+		$username_logged_in = false;
+		$crawler = self::request('GET', 'index.php');
+		$is_logged_in = strpos($crawler->filter('div[class="navbar"]')->text(), 'Login') === false;
+		if ($is_logged_in)
+		{
+			$username_logged_in = $crawler->filter('li[id="username_logged_in"] > div > a > span')->text();
+		}
+		return $username_logged_in;
+	}
+
+	/**
+	 * Disable posting flood control
+	 */
+	protected function disable_flood_interval()
```

Comment:
Currently this is only used in one place. Maybe replace it with a static function that can be called from other tests cases if needed at some point? Also, could you also add a function to enable the flood interval again? Ideally we should aim to try and keep side effects of tests to a minimum.

Reply:
Not sure about static but fixed as I could.